src/System.Linq/src/System/Linq/Partition.cs

```diff
@@ -389,5 +389,308 @@ public int GetCount(bool onlyIfCheap)
                 return Count;
             }
         }
+
+        private sealed class EnumerablePartition<TSource> : Iterator<TSource>, IPartition<TSource>
+        {
+            private readonly IEnumerable<TSource> _source;
+            private readonly int _minIndexInclusive;
```

Comment:
@VSadov I originally had something like that, actually, using ints. But since all of the other partitions used `_minIndexInclusive` and `_maxIndexInclusive` I decided to follow their lead.

> Why not uint _start and uint _length?

I used ints for the fields, since it's more idiomatic. If we know the length (`Take` has been called atl. once) then the count will fit in an int anyway. Also, it'd still be possible that `_start` could overflow during a chained `Skip` / `Take` & we'd have to wrap ourselves in another iterator.